packages/manager/apps/dedicated/client/app/ip/components/list/list.controller.js

```diff
@@ -0,0 +1,636 @@
+import find from 'lodash/find';
+import filter from 'lodash/filter';
+import get from 'lodash/get';
+import map from 'lodash/map';
+import set from 'lodash/set';
+import toInteger from 'lodash/toInteger';
+
+import { IP_TYPE, TRACKING_PREFIX } from './list.constant';
+
+export default class IpListController {
+  /* @ngInject */
+  constructor(
+    $http,
+    $location,
+    $q,
+    $rootScope,
+    $scope,
+    $stateParams,
+    $timeout,
+    $translate,
+    Alerter,
+    Ip,
+    IpFirewall,
+    IpMitigation,
+    IpOrganisation,
+    IpReverse,
+    IpVirtualMac,
+    Validator,
+    atInternet,
+  ) {
+    this.$http = $http;
+    this.$location = $location;
+    this.$q = $q;
+    this.$rootScope = $rootScope;
+    this.$scope = $scope;
+    this.$stateParams = $stateParams;
+    this.$timeout = $timeout;
+    this.$translate = $translate;
+    this.Alerter = Alerter;
+    this.Ip = Ip;
+    this.IpFirewall = IpFirewall;
+    this.IpMitigation = IpMitigation;
+    this.IpOrganisation = IpOrganisation;
+    this.IpReverse = IpReverse;
+    this.IpVirtualMac = IpVirtualMac;
+    this.Validator = Validator;
+    this.atInternet = atInternet;
+  }
+
+  $onInit() {
+    const self = this;
+    const {
+      $http,
+      $location,
+      $q,
+      $rootScope,
+      $scope,
+      $stateParams,
+      $timeout,
+      $translate,
+      Alerter,
+      Ip,
+      IpFirewall,
+      IpMitigation,
+      IpOrganisation,
+      IpReverse,
+      IpVirtualMac,
+      Validator,
+      atInternet,
+    } = this;
+
+    $scope.alerts = {
+      mitigation: [],
+      antihack: [],
+      arp: [],
+      spam: [],
+    };
+    $scope.IP_TYPE = IP_TYPE;
+
+    // pagination
+    $scope.pageNumber = toInteger($stateParams.page) || 1;
+    $scope.pageSize = toInteger($stateParams.pageSize) || 1;
+    $scope.paginationOffset = 1 + ($scope.pageNumber - 1) * $scope.pageSize;
+    if ($scope.pageNumber < 1) {
+      $scope.pageNumber = 1;
+    }
+    if ($scope.pageSize < 10) {
+      $scope.pageSize = 10;
+    } else if ($scope.pageSize > 300) {
+      $scope.pageSize = 300;
+    }
```

Comment:
**question:** Can we refactor and attach on controller object instead of `$scope`?